ocaml/libs/xapi-stdext/lib/xapi-stdext-threads/threadext_test.ml

```diff
@@ -14,22 +14,64 @@
 
 module Delay = Xapi_stdext_threads.Threadext.Delay
 
-let span_approx ~max_error =
-  let eq_within a b =
-    let diff = Mtime.Span.abs_diff a b in
-    Mtime.Span.compare diff max_error < 0
-  in
-  Alcotest.testable Mtime.Span.pp @@ eq_within
-
-let test_wait () =
-  let m = Delay.make () in
-  let c = Mtime_clock.counter () in
-  let time = 1 in
-  let expected = Mtime.Span.(time * s) in
-  let max_error = Mtime.Span.(10 * ms) in
-  let _ = Delay.wait m (float_of_int time) in
-  let wait_time = Mtime_clock.count c in
-  Alcotest.check' (span_approx ~max_error) ~msg:"diff is smaller than max error"
-    ~expected ~actual:wait_time
-
-let () = Alcotest.run "Threadext" [("wait", [("wait", `Quick, test_wait)])]
+let delay_wait_check ~min ~max delay timeout expected =
+  let cnt = Mtime_clock.counter () in
+  let res = Delay.wait delay timeout in
+  let elapsed = (Mtime_clock.count cnt |> Mtime.Span.to_float_ns) *. 1e-9 in
```

Comment:
Changing the type of mix and max to monotonic time spans (Mtime.Span.t) would remove the need for this line.

Then the function can be called with:
```ocaml
delay_wait_check ~min:Mtime.Span.(100 * ms) ~max:Mtime.Span.(110 * ms) d 0.1 true
```

The comparison against elapsed can be done with:
```ocaml
if Mtime.Span.(is_shorter elapsed ~than:min || is_longer elapsed ~than:max) then
    ...
```

Reply:
So you would remove 1 single line to make other 10 much more complicated ? It does not sound much of a gain to me.

Reply:
`Unix` module (`delayf`), `Thread` module (`delay`) and our `Delay` module (`wait`) use seconds for time span so keeping the same time for most of the code makes IMO more consistent than the opposite.

Reply:
Personally, I don't like using floats as a meassure of time because they are unitless.

Reply:
I had this suspicion that the reason was the unitless, but here we are specifically testing an interface that requires time span to be passed as floating point in seconds.

What about specifying the unit for the parameters? Like
```ocaml
delay_wait_check ~min_s:0.1 ~max_s:0.11 d 0.1 true
```
And maybe using `Mtime.Span`, but just inside `delay_wait_check` ?